### PR TITLE
feat: Make use of property index for property constraints

### PR DESCRIPTION
**Pull request by protodef**
Given a property constraint like `{name: {first: 'Tom'}, age: 30}`,
it will be transformed to as below.
```
WHERE properties #>> '{name,first}'::text[] AND
      properties #>> '{age}'::text[]
```
And if one of the target label and its sub-labels has a GIN index on
the `properties` column, it will add the following qualification.
`properties @> jsonb_build_object(...)`

### CODE REVIEW

File: src/backend/parser/parse_graph.c [hunk 1]
```diff
@@ -2598,26 +2616,333 @@ transformElemQuals(ParseState *pstate, Node *qual)
...
+/* See get_relation_info() */
+static bool
+hasGinOnProp(Oid relid)
+{
+	Relation	rel;
+	List	   *indexoidlist;
+	ListCell   *li;
+	bool		ret = false;
+
+	rel = heap_open(relid, NoLock);
```
Comment (mlogue) [inline src/backend/parser/parse_graph.c]: If DDL happens, is it safe to NoLock mode?

File: src/backend/parser/parse_graph.c [hunk 2]
```diff
@@ -2598,26 +2616,333 @@ transformElemQuals(ParseState *pstate, Node *qual)
...
+	ListCell   *li;
+
+	relid = getSourceRelid(pstate, varno, varattno);
+	if (!OidIsValid(relid))
+		return false;
+
+	if (!has_subclass(relid))
+		return hasGinOnProp(relid);
+
+	inhoids = find_all_inheritors(relid, AccessShareLock, NULL);
```
Comment (mlogue) [inline src/backend/parser/parse_graph.c]: inhoids isn't freed

File: src/backend/parser/parse_graph.c (hunk 1 above)
Comment (protodef) [inline src/backend/parser/parse_graph.c]: In `ginAvail()`, `getSourceRelid()` and `find_all_inheritors()` grab `AccessShareLock` of the target relations.

File: src/backend/parser/parse_graph.c (hunk 2 above)
Comment (protodef) [inline src/backend/parser/parse_graph.c]: It doesn't have to be freed.

**Pull request merged**